File: src/main/java/net/zomis/tttultimate/games/TTClassicController.java
package net.zomis.tttultimate.games;

import net.zomis.tttultimate.TTBase;
import net.zomis.tttultimate.TTPlayer;

public class TTClassicController extends TTController {

	public TTClassicController(TTBase board) {
		super(board);
	}

	@Override
	public boolean isAllowedPlay(TTBase tile) {
		return tile.getParent() != null && !tile.hasSubs() &&
				tile.getParent().getWonBy().equals(TTPlayer.NONE) &&
				tile.getWonBy() == TTPlayer.NONE;
	}

	@Override
	public boolean performPlay(TTBase tile) {
		tile.setPlayedBy(currentPlayer);
		tile.getParent().determineWinner();
		nextPlayer();
		return true;
	}

	@Override
	protected void onReset() {
	}

}

File: uttt-common/src/main/kotlin/net/zomis/tttultimate/games/TTClassicController.kt
package net.zomis.tttultimate.games

import net.zomis.tttultimate.TTBase
import net.zomis.tttultimate.TTPlayer

open class TTClassicController(board: TTBase) : TTController(board) {

    override fun isAllowedPlay(tile: TTBase): Boolean {
        return tile.parent != null && !tile.hasSubs() &&
                tile.parent.wonBy.equals(TTPlayer.NONE) &&
                tile.wonBy === TTPlayer.NONE
    }

    override fun performPlay(tile: TTBase): Boolean {
        tile.setPlayedBy(currentPlayer)
        tile.parent!!.determineWinner()
        nextPlayer()
        return true
    }

    override fun onReset() {
    }

}
